### DISCUSSION TITLE
Proxy Auth?

### DISCUSSION THREAD

**Discussion by cinderblockgames** [tiredofit/docker-bookstack#3] [Q&A]
I came to this repo from https://github.com/BookStackApp/BookStack/issues/2223 but I don't see anything in the README about proxy auth.  Is that supported here?

**Comment by tiredofit** [comment]
I am using proxy auth with one of my clients successfully since 2020 (about the time the discussion and PR was made). I don't include the actual code in the Bookstack image, but have my team utilize an override functionality I have built into the image by putting files in /assets/custom which overwrites /www/bookstack files on each container startup. In that are a series of files that handle a) LDAP Synchronization and the header authentication via _SERVER based variables.

Things also changed after initial PR by moving some of the authentication components into the theming component, where it looks something like this in a `functions.php`:

```
Theme::listen(ThemeEvents::WEB_MIDDLEWARE_BEFORE, function (Request $request) {
    $authVarName = config('sso.key');
    $authUserAttribute = config('sso.discover');

    $authVarValue = $_SERVER[$authVarName] ?? null;
    if ($authVarValue && !auth()->check()) {
        $user = User::query()->where($authUserAttribute, '=', $authVarValue)->first();
        if ($user) {
            auth()->login($user, true);
        }
    }
});
```

However this alone does not make it work, one still needs to patch up some files in the /app/Config/Auth folder.
